### ISSUE TITLE
Ongoing Translations

### ISSUE THREAD

**Issue by chaorace** [CQUI-Org/cqui#254]
CQUI is always growing, which means the current localizations are eventually going to fall behind and need revisiting. This is the issue for tracking that. This issue is probably never going to get closed unless we eventually stop adding new strings, so feel free to revisit it occasionally if you'd like to chip in with continuing to keep CQUI as accessible as possible!

Current In-Game Localizations:
* Chinese
* French
* German
* Italian
* Korean
* Polish
* Russian

Languages in need of a new maintainer:
* French
  * LOC_CQUI_BINDINGS
  * LOC_CQUI_BINDINGS_DROPDOWN
  * LOC_CQUI_BINDINGS_DROPDOWN_TOOLTIP
  * LOC_CQUI_CITYVIEW_ITEMHEIGHT
  * LOC_CQUI_GENERAL_RESOURCEDIMMINGSTYLE
  * LOC_CQUI_GENERAL_TOGGLEYIELDSONLOAD
  * LOC_CQUI_GOSSIP
  * LOC_CQUI_GOSSIP_TRIMMESSAGE
  * LOC_CQUI_GOSSIP_TRIMMESSAGE_TOOLTIP
  * LOC_CQUI_LENSES
  * LOC_CQUI_LENSES_AUTOAPPLYARCHAEOLOGISTLENS
  * LOC_CQUI_LENSES_AUTOAPPLYBUILDERLENS
  * LOC_CQUI_LENSES_AUTOAPPLYSCOUTLENS
  * LOC_CQUI_POPUPS
  * LOC_CQUI_POPUPS_ALWAYSOPENTECHTREES
  * LOC_CQUI_POPUPS_TECHVISUAL
  * LOC_CQUI_POPUPS_TECHVISUAL_TOOLTIP
  * LOC_CQUI_POPUPS_TECHAUDIO
  * LOC_CQUI_POPUPS_TECHAUDIO_TOOLTIP
  * LOC_CQUI_CITYVIEW_SMARTWORKICON
  * LOC_CQUI_CITYVIEW_SMARTWORKICON_TOOLTIP
  * LOC_CQUI_CITYVIEW_SMARTWORKICONALPHA
  * LOC_CQUI_CITYVIEW_SMARTWORKICONSIZE
  * LOC_CQUI_CITYVIEW_WORKICONS_HEADER
  * LOC_CQUI_CITYVIEW_WORKICONALPHA
  * LOC_CQUI_CITYVIEW_WORKICONSIZE

**Comment by e1ectron** [comment]
@chaorace if I found some bugs, when I can't add translation, should I create new issue or publish it here?

**Comment by chaorace** [comment]
Feel free to put those here

**Comment by e1ectron** [comment]
# 1. Reports
some strange lines, missing LOCs and don't know where is other things

![1484827583392](https://cloud.githubusercontent.com/assets/5980712/22106304/2fcd7036-de59-11e6-885a-64bf13cd1dcd.png)
---

# 2. Settings

Tooltip on button

<img width="328" alt="2017-01-19 9 06 56" src="https://cloud.githubusercontent.com/assets/5980712/22106434/c9b9cafa-de59-11e6-8641-a53772de8ef0.png">

Not translated dropdowns in General and Bindings, and missing LOC in Gossip.

![1484827634681](https://cloud.githubusercontent.com/assets/5980712/22106362/78027432-de59-11e6-94dd-387bbb7f1fb2.png)
![1484827661016](https://cloud.githubusercontent.com/assets/5980712/22106364/79c334e6-de59-11e6-9b25-aa6a6147a4a0.png)
![1484827680817](https://cloud.githubusercontent.com/assets/5980712/22106369/7b74b170-de59-11e6-8687-f19c4dd97081.png)
---

# 3. Lenses
Cities and Yield lenses legend has hardcoded text.
<img width="190" alt="2017-01-19 9 10 00" src="https://cloud.githubusercontent.com/assets/5980712/22106457/e2459b1c-de59-11e6-8c5f-fa38e3d85b1f.png">
<img width="194" alt="2017-01-19 9 10 51" src="https://cloud.githubusercontent.com/assets/5980712/22106462/e652c34c-de59-11e6-82fc-f245c4ef7d73.png">
---

# 4. Great People
After contraction of Great People's cards, russian translate of unknown player overlap data on the right.
<img width="677" alt="2017-01-19 9 12 15" src="https://cloud.githubusercontent.com/assets/5980712/22106537/2669f7a2-de5a-11e6-8de7-4d3ec3eadf21.png">

**Comment by sejbr** [comment]
Few bugs/overlooked translations:

# 1. City view
Translated lines don't apply in the city view, also was "Water" in the base game? Can't remember. If it wasn't then there's no string to translate it.
![1](https://cloud.githubusercontent.com/assets/6895093/22112776/8d1d7c26-de64-11e6-8a89-add1d6dd5fbd.jpg)

# 2. Trade view aligning
![2](https://cloud.githubusercontent.com/assets/6895093/22112778/8f879640-de64-11e6-889d-64148f9c49f5.jpg)

**Comment by chaorace** [comment]
Okay, so the lenses and trade screen are part of @astog 's mods, so my first line of action for dealing with those cases is passing the info on to him (same thing for Greg Miller's report screen mod, but he's proven a lot more elusive!). My first priority is going to be fixing the LOC issues in CQUI proper, then addressing the integrations.

That being said, here's my plan for those:
* The tooltip has already been removed. I figured there wasn't much point in that specific tooltip
* I'll be adding LOCs for the dropdowns
* I'll see about findinga replacement LOC that's already in-game for that missing title in the gossip menu
* I'll add text wrapping to the those instances in the GP panel, as well as using the infinity symbol instead of the abbreviation
* I'll also add those missing LOCs back to the city details panel

The above changes are now my Immediate priority, so expect the above to be done sometime after I get home today

**Comment by chaorace** [comment]
Okay: added dropdown LOCs, replaced Diplo LOC which should now work in Russian, and [added text wrapping to the instances in the GP panel](http://i.imgur.com/qyVwtBY.png) (as well as some restructuring and using a language-neutral infinity)

The city details panel LOC fixes are going to take some time since it looks like it's going to take a lot of rewriting, will continue work on that tomorrow. Expect the commit containing the above fixes to go live soon.

The following new LOC strings have been added:
LOC_CQUI_GENERAL_HIDDEN
LOC_CQUI_GENERAL_SOLID
LOC_CQUI_GENERAL_TRANSPARENT

**Comment by chaorace** [comment]
Alright, the LOCs have been fixed for the CityPanelOverview. The following LOCs have been obsoleted and are now removed from the LOC data:
* LOC_CQUI_CITY_GREAT_PEOPLE
* LOC_CQUI_CITY_NATIONAL_PARK
* LOC_CQUI_CITY_WAR_WEARINESS

**Comment by chaorace** [comment]
New LOC strings have been added to CQUI_Text_Settings:
* LOC_CQUI_UNITS
* LOC_CQUI_UNITS_AUTOEXPANDUNITACTIONS

**Comment by EvolSknaht** [comment]
In   #262  following LOCs have been added 
In [BTS_Text.xml](*)
- LOC_TRADE_GROUP_SETTINGS    
- LOC_TRADE_FILTER_SETTINGS

In [URS_Text.xml](*)
- LOC_HUD_REPORTS_TAB_CURRENT_DEALS
- LOC_HUD_REPORTS_TAB_HIDE_CITY_BUILDINGS
- LOC_CURRENT_DEALS_DEAL_WITH
- LOC_CURRENT_DEALS_ENDS_IN
- LOC_CURRENT_DEALS_ENDS_TURNS
- LOC_CURRENT_DEALS_OUTGOING
- LOC_CURRENT_DEALS_INCOMING
- LOC_CURRENT_DEALS_NO_ROUTE
- LOC_CURRENT_DEALS_NO_YIELDS
- LOC_CURRENT_DEALS_NONE_TOTALS
- LOC_CURRENT_DEALS_TOTALS
- LOC_UNITS_MILITARY_LAND
- LOC_UNITS_MILITARY_AIR
- LOC_UNITS_MILITARY_SEA
- LOC_UNITS_CIVILIAN_AND_SUPPORT
- LOC_UNITS_TYPE
- LOC_UNITS_VIEW_ON_THE_MAP
- LOC_UNITS_NUMBER_OF_UNITS

And the LOC_TRADE_COLLAPSE_ALL_BUTTON_TEXT
in [BTS_Text.xml](*) need to be shorter to show correctly.
![should_be_short](https://cloud.githubusercontent.com/assets/18080141/22219416/a62895aa-e1e5-11e6-89a0-01eafda1ce14.PNG)
All these above finished:
  - [x] Chinese
  - [ ] Italian
  - [ ] Russian
  - [ ] Polish

**Comment by chaorace** [comment]
The following new LOCs have been added:

* CQUI_Text_General:
  * LOC_CQUI_GREAT_PERSON_PROGRESS
* BTS_Text:
  * LOC_TRADE_GROUP_SETTINGS
  * LOC_TRADE_FILTER_SETTINGS
* URS_Text (NEW FILE!):
  * LOC_HUD_REPORTS_TAB_CURRENT_DEALS
  * LOC_HUD_REPORTS_TAB_HIDE_CITY_BUILDINGS
  * LOC_CURRENT_DEALS_OUTGOING
  * LOC_CURRENT_DEALS_INCOMING
  * LOC_UNITS_MILITARY_LAND
  * LOC_UNITS_MILITARY_AIR
  * LOC_UNITS_MILITARY_SEA
  * LOC_UNITS_VIEW_ON_THE_MAP

**Comment by e1ectron** [comment]
I have a little idea:
what if we get all LOCs we have and paste it to Google Sheet. [Example](https://docs.google.com/spreadsheets/d/1jDO0EHKIaO-Vbn8pZpWQf2HIGKSD055regi6XdxkvAk/edit?usp=sharing)
And here we can fast and easy find, what is missing.
Other people, who use mod, don't know how to use git, but have a lot of language experience can comment this table or offer new language translate.

I understand that you are very busy developing and supporting issues, so that it can be an unwanted chore.
But we can add editor rights to people, who make a regular translations, and add the rule — change table only after Pull request accepting.

**Comment by chaorace** [comment]
I think that's a very good idea. I'll set one up now!

**Comment by chaorace** [comment]
Alright, the sheet is live here: 

https://docs.google.com/spreadsheets/d/1Xf8NnhO7TRN_urweUeaFenGtL5GInNSmgb8JyUFHjuM

@e1ectron @EvolSknaht @sejbr Please request edit permissions from within the Google Sheet

**Comment by sejbr** [comment]
Pull requested updated polish localization.
@EvolSknaht is 4 letter "Rozw/Zwiń" acceptable for the LOC_TRADE_COLLAPSE_ALL_BUTTON_TEXT?
Didn't have time to test it. Hope it fits.
@chaorace I'll edit the google doc if needed in spite of github commit.

**Comment by deggesim** [comment]
Today or tomorrow I update Italian localization ☺

**Comment by deggesim** [comment]
I updated Google Sheet with Italian updated localization.
As you can see modinfo section is not localized because sometimes technical words are not localized.

**Comment by chaorace** [comment]
The following new LOCs have been added:

* LOC_CQUI_CITYVIEW_SMARTBANNER_HEADER
* LOC_CQUI_CITYVIEW_SMARTBANNER_UNLOCKEDCITIZEN_TOOLTIP
* LOC_CQUI_CITYVIEW_SMARTBANNER_DISTRICTS_TOOLTIP
* LOC_CQUI_CITYVIEW_SMARTBANNER_POPULATION_TOOLTIP
* LOC_CQUI_CITYVIEW_SMARTBANNER_CULTURAL_TOOLTIP

**Comment by chaorace** [comment]
The text of LOC_CQUI_CITYVIEW_SMARTBANNER_TOOLTIP has also changed somewhat and will need updating now

Keep in mind that LOC_CQUI_CITYVIEW_SMARTBANNER_POPULATION_TOOLTIP is likely to change in the near future, see more info in the discussion of #255

**Comment by Riskke** [comment]
Hey there! I'm new to github, no idea how this work but I'd love to help translate this amazing mod to Spanish. Please tell me how can I get the files for editing and later on sharing them

**Comment by e1ectron** [comment]
Hey, @Riskke!
You can see all texts in [google docs file](https://docs.google.com/spreadsheets/d/1Xf8NnhO7TRN_urweUeaFenGtL5GInNSmgb8JyUFHjuM),
inlcude archive with translate here and later we add it. 
Important to add LOC name (ex. "CQUI_TEASER") near of translate text.

Or, maybe, @chaorace have a better idea.

**Comment by chaorace** [comment]
@Riskke I've added you as an editor to the google doc, thanks for contributing!

**Comment by chaorace** [comment]
The following new LOCs have been added:
* LOC_CQUI_GENERAL_MINIMAPSIZE
* LOC_CQUI_POPUPS_ALWAYSOPENTECHTREES

I'll get around to updating the list at the top of the thread and on the docs tomorrow when I'm less tired. Good night folks!

**Comment by chaorace** [comment]
O-Kay, time to tackle updating these lists

**Comment by chaorace** [comment]
The following translations have been added in3b26693619c99f15628d4201a5914fcfb057fec3:

LOC_CQUI_CITYVIEW_SMARTWORKICON
LOC_CQUI_CITYVIEW_SMARTWORKICON_TOOLTIP
LOC_CQUI_CITYVIEW_SMARTWORKICONALPHA
LOC_CQUI_CITYVIEW_SMARTWORKICONSIZE
LOC_CQUI_CITYVIEW_WORKICONS_HEADER
LOC_CQUI_CITYVIEW_WORKICONALPHA
LOC_CQUI_CITYVIEW_WORKICONSIZE

**Comment by wbqd** [comment]
Hello! I would like to add a Korean translation. Can I get permission to edit this spreadsheet? I have already sent permission requests.

**Comment by chaorace** [comment]
@wbqd I've added access for you on the google doc,, feel free to make a new page in the doc if you prefer

**Comment by wbqd** [comment]
I've finished translation on the google doc, BTW I've found a bug:

![civ_bug](https://cloud.githubusercontent.com/assets/3282649/23092631/a664e21c-f612-11e6-878e-e36f521ce81d.PNG)

**Comment by sejbr** [comment]
#310 
PR'ed the polish localization update. I've also updated the google doc with loc strings and polish+english translations since those were missing for the new lines.

**Comment by sejbr** [comment]
#310

**Comment by wbqd** [comment]
@chaorace, This part of LOC has already been updated in the source code in the last PR. I've also made changes to the Google Docs now.

**Comment by Riskke** [comment]
Done with the Spanish translation. Please msg me if any errors.

Keep the amazing work!

**Comment by chaorace** [comment]
@Riskke Excellent! Once the post-patch chaos calms down a bit I'll work on porting your translations over to the LOC files!

**Comment by sejbr** [comment]
@chaorace finally had some time to start a new game and found that some settings don't fit their boxes. Should I try end shorten the translations to fit them or would you be able to fit boxes to text?

What I'm talking about:
![image](https://cloud.githubusercontent.com/assets/6895093/23334625/fb5d79ce-fba2-11e6-8d0a-f986d0776ec4.png)

**Comment by e1ectron** [comment]
@sejbr I met this problem too several times and decided make translate shorter. Often this decision makes my translate more correct than word by word.

**Comment by deggesim** [comment]
Me too, I use the same solution.

**Comment by sejbr** [comment]
Whoops, accidentally added the comment to my fork and spammed here a bit. Any way to remove those references? Updating commit message doesn't seem to remove them.

**Comment by chaorace** [comment]
@e1ectron @sejbr @EvolSknaht  @deggesim  @wbqd Check your emails, folks, you're all invited to the CQUI Language team!

**Comment by e1ectron** [comment]
@chaorace great! now I can clone repo and push loc changes right this repo, or I still need forks and do pull requests?

**Comment by Riskke** [comment]
Get me in the club for the Spanish translation ^^

Spring patch coming soon anyways

**Comment by JHCD** [comment]
Content of LOC_CQUI_SETTINGS_REMINDER will change with the PR #447:

>Remember: CQUI settings are attached to individual saves![NEWLINE]If you'd like to change a setting by default, please edit the CQUI_Settings.sql file within the [MOD INSTALL DIRECTORY]/CQUI/Assets directory

Language changed:
- [x] Englisch
- [ ] Chinese
- [ ] French
- [x] German
- [ ]     Italian
- [ ]     Korean
- [ ]     Polish
- [ ]     Russian

**Comment by JHCD** [comment]
There will be two new LOCs with the PR #488:

>    LOC_CQUI_GENERAL_BLOCKONCITYATTACK
    LOC_CQUI_GENERAL_BLOCKONCITYATTACK_TOOLTIP

Language changed:

- [x] Englisch
- [ ] Chinese
- [ ] French
- [x] German
- [ ]     Italian
- [ ]     Korean
- [ ]     Polish
- [ ]     Russian

**Comment by chaorace** [comment]
I'm closing this issue, as language discussion is being moved to our Slack channel

**Comment by e1ectron** [comment]
@chaorace where I can find a link to channel?

**Comment by chaorace** [comment]
@e1ectron I need an email to invite you, please send me an email at chaorace@gmail.com to get it

**Comment by e1ectron** [comment]
@chaorace already sent it 22.05 with "slack channel" title.
